app/policies/case/sar_policy.rb

```diff
@@ -39,6 +39,16 @@ def resolve
 
   end
 
+  def respond_and_close?
+    clear_failed_checks
+    self.case.drafting? &&
+        user.responding_teams.include?(self.case.responding_team)
+  end
+
+  def process_respond_and_close?
+    respond_and_close?
```

Comment:
Can we just have the controller actions involved call the same policy? Less code that way, unless there's a rationale to not.